poptorch_experimental_addons/_impl/collectives.py

```diff
@@ -35,4 +35,32 @@ def all_gather_cross_replica_mean_grad(x: torch.Tensor, replication_factor: int)
     return out
 
 
-__all__ = ["all_gather_cross_replica_mean_grad"]
+def all_reduce_cross_replica_sum(
+    x: torch.Tensor, replication_factor: int, insert_in_grad_graph: bool = False
+) -> Any:
+    """
+    All-reduce across IPU program replicas
+
+    Sums tensors occupying the same memory location across IPUs, resulting
+    in replicated tensors.
+
```

Comment:
Worth explaining `insert_in_grad_graph`, I think, although it is a very good name.